sycl/source/detail/memory_manager.cpp

```diff
@@ -266,8 +266,8 @@ void copyD2H(SYCLMemObjI *SYCLMemObj, RT::PiMem SrcMem, QueueImplPtr SrcQueue,
     if (1 == DimDst && 1 == DimSrc) {
       PI_CALL(RT::piEnqueueMemBufferRead(
           Queue, SrcMem,
-          /*blocking_read=*/CL_FALSE, DstOffset[0], DstAccessRange[0],
-          DstMem + DstOffset[0], DepEvents.size(), &DepEvents[0], &OutEvent));
+          /*blocking_read=*/CL_FALSE, SrcOffset[0], DstAccessRange[0],
+          DstMem, DepEvents.size(), &DepEvents[0], &OutEvent));
```

Comment:
I believe that you still need to adjust DstMem by DstOffset.

Reply:
Restored